core/src/transport.rs

```diff
@@ -54,6 +55,8 @@ pub use self::memory::MemoryTransport;
 pub use self::optional::OptionalTransport;
 pub use self::upgrade::Upgrade;
 
+static NEXT_LISTENER_ID: AtomicU64 = AtomicU64::new(1);
```

Comment:
Should we change this to `AtomicUsize` and use usize in `ListenerId` instead?

Reply:
I don't think it really matters, happy to leave it as a u64. What do you think?

Reply:
I think we can leave it

Reply:
Adding to the above the argument of portability.

> PowerPC and MIPS platforms with 32-bit pointers do not have AtomicU64 or AtomicI64 types.

https://doc.rust-lang.org/std/sync/atomic/index.html#portability

Thus slight preference for `AtomicUsize` from my end. Also consistent with `ConnectionId`.